rskj-core/src/main/java/org/ethereum/rpc/JsonRpcWeb3FilterHandler.java

```diff
@@ -9,42 +9,80 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Created by ajlopez on 18/10/2017.
  */
 
 @ChannelHandler.Sharable
 public class JsonRpcWeb3FilterHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
-    private static final Logger LOGGER = LoggerFactory.getLogger("jsonrpc");
+    private static final Logger logger = LoggerFactory.getLogger("jsonrpc");
+    private final List<String> rpcHost;
+    private final InetAddress rpcAddress;
 
     private OriginValidator originValidator;
 
-    public JsonRpcWeb3FilterHandler(String corsOrigins) {
-        this.originValidator = new OriginValidator(corsOrigins);
+    public JsonRpcWeb3FilterHandler(String corsDomains, InetAddress rpcAddress, List<String> rpcHost) {
+        this.originValidator = new OriginValidator(corsDomains);
+        this.rpcHost = rpcHost;
+        this.rpcAddress = rpcAddress;
     }
 
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws Exception {
-        HttpMethod httpMethod = request.getMethod();
         HttpResponse response;
+        HttpMethod httpMethod = request.getMethod();
+        HttpHeaders headers = request.headers();
+        String hostHeader = headers.get(HttpHeaders.Names.HOST).split(":")[0];

```

Comment:
The `headers.get` method returns the value of the header, then the split is spliting the host name that should not have `:` however IPv6 addresses have `:`

Reply:
I'll have to check this better since somehow host header is including port